cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis/ListMap.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.CLI.Shelley.Run.Genesis.ListMap
+  ( ListMap(..)
+  ) where
+
+import Cardano.CLI.Shelley.Run.Genesis.LazyToJson (LazyToJson(..))
+import Data.Aeson (Value(..), ToJSON(..), ToJSON1(..), ToJSON2(..), ToJSONKey(..), ToJSONKeyFunction(..))
+import Data.Aeson.Encoding ( dict )
+import Data.Aeson.Types (listValue)
+import Data.Eq (Eq(..))
+import Data.Function ((.), id)
+import Data.Functor ((<$>))
+import Data.Monoid (Monoid(..))
+import Data.Semigroup (Semigroup(..))
+import Prelude (uncurry)
+import Text.Show (Show(..))
+
+import qualified Data.Aeson as J
+import qualified Data.Aeson.Encoding as E
+import qualified Data.Aeson.KeyMap as KM
+import qualified Data.ByteString.Builder as B
+import qualified Data.List as L
+import qualified Data.Vector as V
+
+newtype ListMap k v = ListMap
```

Comment:
We should be able to generically derive the JSON instances for `ListMap` and it should do the right thing. @newhoggy can you try this?

Reply:
Note, generically deriving JSON would do one of two things:

`deriving newtype` would change the format of the JSON file to use `[["key1", "value1"], ["key2", "value2"]]` instead of `{"field1": "value1", "field2": "value2"}`.

`DeriveGeneric` would additionally add `{ "unListMap": ... }` wrapper.

Reply:
Looks like we're settling on changing the ledger code, so this code will disappear once the ledger has been updated.